CONTRIBUTING.md

```diff
@@ -120,14 +122,237 @@ rework your branch. You should rework any relevant information from the GitHub
+  commit rather than being spread across four commits.
+
+- Force-push the newly updated branch to your fork on GitHub.
+
+  This will magically update your PR with your new (*single) commit.
+
+Taking each step in turn:
+
+- Start the rebase operation:
+
+  ```sh
+  $ git rebase -i master
+  ```
+
+  Your editor window will now show something like:
+
+  ```
+  pick 2e335ac docs: Fix typos and fold long lines
+  pick 6d6deb0 apply review feedback changes
+  pick 23bc01d remove blank lines
+  pick 3a4ba3f doh - missed one
+  ```
+
+- In your editor, read the comments at the bottom of the screen. Next, without
+  modifying the first line (`pick 2e335ac docs: Fix typos and fold long
+  lines`), change the "`pick`" at the start of all the other lines to
+  "`squash`".
+
+  Your editor window will now show something like:
+
+  ```
+  pick 2e335ac docs: Fix typos and fold long lines
+  squash 6d6deb0 apply review feedback changes
+  squash 23bc01d remove blank lines
+  squash 3a4ba3f doh - missed one
+  ```
+
+  Now save and quit the editor window. Git will then put you *back* into your
+  editor. This time, rather than showing you all the one line commit
+  summaries, it will show you all the commit *messages* (all the descriptions
+  for all the commits you created). Git is asking you to modify the file as
+  you wish. When you save and exit the editor, git will use
+  whatever was left in the file as the commit message for your "squashed"
+  commit. Assuming you followed the [example first PR](#example-first-pr),
+  your first commit ("`2e335ac docs: Fix typos and fold long lines`") will
+  already be in the [correct format](#patch-format). Therefore all you need to
+  do is keep the text for your first commit and delete everything else in the
+  editor window, updating it if appropriate based on the review feedback.
+
+  Save the file and quit the editor. Once this operation completes, the four
+  commits will have been converted into a single new commit. You can check this
+  by running the `git log command again:
+
+  ```sh
+  $ git log master.. --oneline --decorate=no
+  3ea3aef docs: Fix typo
+  ```
+
+- Force push your updated local `fix-doc-bugs` branch to your remote fork:
+
+  ```sh
+  $ git push -f github
+  ```
+
+  > **Note:** The `-f` option is the magic "force push". Since you have
+  > created a new commit using `git rebase`, you need to "overwrite" the old
+  > copy of your branch in your fork on GitHub. This might sound drastic but
+  > its similar to saving a file in your editor - you are simply overwriting
+  > the old version that you no longer want. Force pushing is the GitHub
+  > recommended way to handle an update to your PR - it also means you do not
+  > have to create a new PR on GitHub; force pushing will effectively re-use
+  > the initial PR you raised, but will include your latest changes.
+
+  For more information on GitHub "force push" workflows see "[Why and how to
```

Comment:
Did you intend to indent this paragraph, but not make it a '>' comment type - just noting that is unusual in our project doc style.